lua/bufferline/jump_mode.lua

```diff
@@ -110,16 +116,10 @@ function JumpMode.unassign_letter(letter)
   JumpMode.reinitialize = true
 end
 
---- @param bufnr integer
---- @return string letter assiegned to `bufnr`
-function JumpMode.get_letter(bufnr)
-  return JumpMode.letter_by_buffer[bufnr] or JumpMode.assign_next_letter(bufnr)
-end
-
 --- Unassign the letter which is assigned to `bufnr.`
 --- @param bufnr integer
 function JumpMode.unassign_letter_for(bufnr)
-  JumpMode.unassign_letter(JumpMode.get_letter(bufnr))
+  JumpMode.unassign_letter(JumpMode.letter_by_buffer[bufnr])
```

Comment:
The function `unassign_letter` returns if `get_letter(bufnr)` returns `nil`. However, `get_letter` has a special case where if `letter_by_bufnr[bufnr]` is `nil`, it will assign a _new_ letter for `bufnr`. This leads to a case where, if we try to unassign a `bufnr` which isn't already assigned, it will always unnecessarily assign the `bufnr`  first, just so it can then immediately undo that action.